src/fundamental/macro-fundamental.h

```diff
@@ -327,14 +327,23 @@ static inline size_t ALIGN_TO(size_t l, size_t ali) {
         return ((l + ali - 1) & ~(ali - 1));
 }
 
+#define ALIGN2(l) ALIGN_TO(l, 2)
 #define ALIGN4(l) ALIGN_TO(l, 4)
 #define ALIGN8(l) ALIGN_TO(l, 8)
+#define ALIGN2_PTR(l) ((void*) ALIGN_TO((uintptr_t) l, 2))
+#define ALIGN4_PTR(l) ((void*) ALIGN_TO((uintptr_t) l, 4))
+#define ALIGN8_PTR(l) ((void*) ALIGN_TO((uintptr_t) l, 8))
```

Comment:
```C
#define ALIGN2_PTR(l) ((void*) ALIGN2((uintptr_t) l)
...
```
?
This makes it clearer that this is just about casting the types back and forth.